packages/common/src/types.ts

```diff
@@ -142,11 +150,13 @@ export interface PropPanel<T extends Schema> {
  * @property {function} pdf Function for rendering PDFs.
  * @property {function} ui Function for rendering UI.
  * @property {PropPanel} propPanel Object for defining the property panel.
+ * @property {function} [shouldRerenderVars] Function for determining which variables should be dependencies of the useEffect deciding whether to rerender the schema in the UI
  */
 export type Plugin<T extends Schema & { [key: string]: any }> = {
   pdf: (arg: PDFRenderProps<T>) => Promise<void> | void;
   ui: (arg: UIRenderProps<T>) => Promise<void> | void;
   propPanel: PropPanel<T>;
+  shouldRerenderVars?: (arg: RerenderCheckProps) => any[];
```

Comment:
I couldn't find a better way to do this. 

When we are typing into the box we are assessing the content to see if a new variable needs to be created, which populates a variable box in the Right Details section. This requires an onChange to the schema, but by default this will lose focus whilst you are typing. 

I tried the value itself as one of the change factors on the `UseEffect` but this broke removing an image on the image schema.

Allowing the schema to determine whether it should re-render seemed like the best option

Reply:
@peteward 
I see that this value is being used as a dependency array in useEffect.
I believe it was added here after trying various things, but I wonder if it's difficult to achieve the same thing without adding this property by modifying or changing the image schema?

It's not that I'm particularly averse to adding new properties to the plugin type. In fact, I've recently come to think that icon and multilingual labels should be here, so I welcome the modification.

However, regarding`shouldRerenderVars`, I feel it's a complicated specification intended for people who understand the internal structure, so I'd prefer not to add it if possible.

Reply:
Yeah I know, it's not ideal.

I will try again

Reply:
Hi @hand-dot ,

I couldn't find a way to make it work for all scenarios without some configuration of the plugin. In my case there are changes to both the `value` (the text) and the `schema` (the variables) which need to be ignored when editing. 

For text entry plugins, if you are typing then you do not want the field to be rerendered until you leave edit mode. However, this is not the case for other schemas such as image and signature. 

I managed to simplify it so that instead of a function it's just a boolean which you can set on the plugin. It might be an advanced option, but it might help other plugin authors in future:

https://github.com/pdfme/pdfme/pull/502/commits/33c8992282f5f2f5cebadd5d54ba9b554e519dbb

Let me know if you think this is better.